pkg/cluster/config.go

```diff
@@ -354,24 +356,22 @@ func IsCloudControllerOwnedByCCM(oclient configclient.Interface) (bool, error) {
 	return ownedByCCM, nil
 }
 
-// getProxyVars returns a map of the cluster-wide proxy variables and values.
-// These are injected by OLM automatically to all its managed operators.
-func getProxyVars() map[string]string {
-	// proxyEnvVars is a list of the supported proxy-related environment variables
-	proxyEnvVars := []string{"HTTP_PROXY", "HTTPS_PROXY", "NO_PROXY"}
+// IsProxyEnabled returns whether a global egress proxy is active in the cluster
+func IsProxyEnabled() bool {
+	return len(ClusterWideProxyVars) > 0
+}
 
-	proxyVars := make(map[string]string, len(proxyEnvVars))
-	for _, envVar := range proxyEnvVars {
-		// read settings from the WMCO container's environment
+// getProxyVars returns a map of the proxy variables and values from the WMCO container's environment. The presence of
+// any implies a proxy is enabled, as OLM would have injected them into the operator spec. Returns an empty map otherwise.
+func getProxyVars() map[string]string {
```

Comment:
Please make this a getter, which will return `ClusterWideProxyVars` instead of having the user access the variable directly. If isn't already cached the variable can be populated before returning it, and the init function can be removed.

Using a getter with a documented function provides a better interface for a user of this package.

Init functions cannot be error handled, and I believe we should avoid them where possible.